### PR TITLE
Update samples to use env: flex

### PR DESCRIPTION
**Pull request by remi**
Additional Updates needed:
* Update Ruby Client Library calls in samples
* Update **entrypoint**: bundle exec rackup -p $PORT
* Update Google Managed VMs in READMEs
* Update supported Ruby versions in READMEs

 - [x] Hello World
 - [x] Cloud SQL
 - [x] Cloud Datastore
 - [x] Cloud Storage
 - [x] Google Authentication
 - [x] Logging
 - [x] Task Queueing

### CODE REVIEW

**Comment by frankyn** [comment]
PTAL @remi

File: 2-cloud-datastore/app/controllers/books_controller.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@ def destroy
   end
   # [END destroy]
 
-  before_filter :convert_published_on_to_date
+  before_action :convert_published_on_to_date
```
Comment (remi) [inline 2-cloud-datastore/app/controllers/books_controller.rb]: Will this work with Rails 4 ~ or was this a Rails 5 update to revert?

File: 2-cloud-datastore/config/environments/development.rb [hunk 2]
```diff
@@ -29,7 +29,7 @@
   # Print deprecation notices to the Rails logger.
   config.active_support.deprecation = :log
 
-  config.serve_static_files = true
+  config.public_file_server.enabled = true
```
Comment (remi) [inline 2-cloud-datastore/config/environments/development.rb]: Rails 5 or is this OK in Rails 4?

File: 2-cloud-datastore/app/controllers/books_controller.rb (hunk 1 above)
Comment (frankyn) [inline 2-cloud-datastore/app/controllers/books_controller.rb]: I missed these during my revert. Thanks!

**Comment by frankyn** [comment]
PTAL @remi

**Pull request merged**